src/zabapgit_gui_router.prog.abap

```diff
@@ -171,6 +174,21 @@ CLASS lcl_gui_router IMPLEMENTATION.
       WHEN gc_action-repo_toggle_fav.                 " Toggle repo as favorite
         lcl_services_repo=>toggle_favorite( lv_key ).
         ev_state = gc_event_state-re_render.
+      WHEN gc_action-repo_transport_to_branch.
+        ev_state = gc_event_state-re_render.
+
+        lt_transport_headers = lcl_transport_popup=>show( ).
+        lt_tadir = lcl_transport=>to_tadir( lt_transport_headers ).
+        IF lt_tadir IS INITIAL.
+          lcx_exception=>raise( 'Canceled or List of objects is empty ' ).
+        ENDIF.
+        ls_transport_to_branch = lcl_popups=>popup_to_create_transp_branch(
+          it_transport_headers = lt_transport_headers
+          it_transport_objects = lt_tadir ).
+        lcl_services_repo=>transport_to_branch(
+          iv_repository_key      = lv_key
+          is_transport_to_branch = ls_transport_to_branch
+          it_transport_objects   = lt_tadir ).
```

Comment:
can we move these 20 lines to lcl_services_repo=>transport_to_branch? would like to keep the router as clean as possible